src/DurableTask.SqlServer/Scripts/logic.sql

```diff
@@ -1060,7 +1061,8 @@ BEGIN
     FROM
         Instances I
     WHERE
-        I.[TaskHub] = @TaskHub AND
+        (@TaskHubFilter IS NULL AND I.[TaskHub] = @TaskHub OR 
+            I.[TaskHub] IN (SELECT [value] FROM string_split(@TaskHubFilter, ','))) AND
```

Comment:
We don't normally allow the caller to specify any task hub information to the stored procedures. This is intentional to create a degree of security isolation between task hubs, which might be owned by different tenants. The task hub is typically determined by the connection string, which helps enforce this isolation, particularly when the task hub name comes from the login (which is unique for each tenant).

Allowing callers to provide a task hub filter that overrides the task hub inferred from the connection could allow one tenant to query information belonging to another tenant. For this reason, I'm not comfortable supporting this change.

Reply:
Thank you for this remark. I wasn't sure myself this filtration is necessary, since there are no other examples of filtering by hub name which is not current one for caller.
Removed filtration logic, and throwing `NotSupportedException` now if filter is present.